src/background.js

```diff
@@ -173,36 +173,30 @@ const messageOps = {
 function syncBookmarks(items) {
   const title = browser.i18n.getMessage('bookmarkFolderTitle');
   browser.bookmarks.search({title: title}).then(folders => {
-    return Promise.all(folders.map(folder => {
-      return browser.bookmarks.removeTree(folder.id);
-    }));
-  }).then(() => {
+    if (folders.length) {
+      return folders[0];
+    }
     return browser.bookmarks.create({title: title});
   }).then(snoozeTabsFolder => {
     log('Sync Folder!', snoozeTabsFolder, Object.values(items));
-    const sortedEntries = [...Object.values(items)];
-    sortedEntries.sort((a, b) => {
-      if (a.time === NEXT_OPEN) {
-        if (b.time === NEXT_OPEN) {
-          return -(a.title.localeCompare(b.title));
-        } else {
-          return 1;
-        }
-      } else if (b.time === NEXT_OPEN) {
-        return -1;
-      } else {
-        return b.time - a.time;
-      }
-    });
+    return browser.bookmarks.getChildren(snoozeTabsFolder.id).then((bookmarks) => {
+      const tabs = [...Object.values(items)];
+      const toCreate = tabs.filter((tab) => !bookmarks.find((bookmark) => tab.url === bookmark.url));
+      const toRemove = bookmarks.filter((bookmark) => !tabs.find((tab) => bookmark.url === tab.url));
 
-    return Promise.all(sortedEntries.map(item => {
-      return browser.bookmarks.create({
-        parentId: snoozeTabsFolder.id,
-        title: item.title,
-        url: item.url,
-        index: 0
-      });
-    }));
+      const operations = toCreate.map(item => {
+        log(`Creating ${item.url}.`);
+        return browser.bookmarks.create({
```

Comment:
The practical impacts are:

### For users that don't use Sync

One bundle of SQL writes and appropriate `fsync`s for each closed and each opened tab. Potential churn in any caches that Places maintains. This is not good for perf, but perhaps users don't snooze tabs often enough for them to notice.

### For users that do use Sync

A sync will start as soon as the first change is written, and will continue contending with this code. Syncs will continue to run (probably only one more) for so long as writes in this loop have occurred during a sync. This will be janky as hell, but #291 will avoid the risk of data loss due to competing devices syncing.

If you're flushing tabs frequently, you'll be not only causing potential jank from hitting the filesystem, but also by causing one or two syncs every time you flush.

The sync will also spin the event loop, potentially causing surprising behavior, such as UI elements losing focus. It will definitely test any race conditions in your code!